src/main/java/org/example/blockchain/Blockchain.java

```diff
@@ -0,0 +1,59 @@
+package org.example.blockchain;
+
+import lombok.Getter;
+import org.example.exception.InvalidMiningException;
+import org.example.exception.PreviousHashMismatchException;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.example.util.StringUtil.mineBlock;
+
+@Getter
+public class Blockchain {
+    private final List<Block> blocks = new LinkedList<>();
```

Comment:
is LinkedList better then ArrayList here?

Reply:
@ani-hyan 
Regarding this question, I asked because we should use LinkedList in cases when we have a big amount of insertions and less amount of random read operations, so you should decide based on this rule
P. S. Generally, ArrayList is preferable, but it depends on you case